### PR TITLE
Fixes to Code::Blocks compilation

### PR DESCRIPTION
**Pull request by DisherProject**
I recently tried to compile Wesnoth using Code::Blocks and i got stuck due to a couple fo issues. 
Anyway, i'm not completely sure that replacing "strnlen" in this way is the best chioce.

### CODE REVIEW

File: src/hotkey/command_executor.cpp [hunk 1]
```diff
@@ -556,7 +556,7 @@ static void event_execute( const SDL_Event& event, command_executor* executor)
 		if(!CKey::is_uncomposable(event.key) && !(mods & KMOD_CTRL) && !(mods & KMOD_ALT) && !(mods & KMOD_GUI)) {
 				return;
 		}
-	} else if(event.type == SDL_TEXTINPUT && strnlen(event.text.text, 32) != 1) {
+	} else if(event.type == SDL_TEXTINPUT && strlen(event.text.text) != 1) {
```
Comment (GregoryLundberg) [inline src/hotkey/command_executor.cpp]: Use strnlen. Either upgrade your compiler or write a shim for your personal use.

**Comment by Vultraz** [comment]
FTR I also get an error with that goddamn `strnlen` case. No idea why.

**Comment by GregoryLundberg** [comment]
http://en.cppreference.com/w/c/string/byte/strlen

try strnlen_s()

**Comment by Vultraz** [comment]
Nope, still says not declared in scope, even with `<cstring>` included.

**Comment by GregoryLundberg** [comment]
https://stash.desy.de/projects/CRYS/repos/crystfel/browse/lib/strnlen.c?at=39d539d1709026cae9af57284c8a0ab5b73a6de3

Sounds like you need to shim it in.

**Comment by Vultraz** [comment]
Could you explain why that damn check is even there?

**Comment by GregoryLundberg** [comment]
To me it reads:

"if this is modified key, such as Ctrl+A or Alt-F1, or a single-byte 'text' character, then execute the hotkey command."

I imagine the test for a single-byte character is to exclude things like a multi-byte Chinese glyph or a multi-byte Unicode code-point.

An equivalent test would be to replace 
`    strnlen(event.text.text, 32) != 1`
with
`    ((event.text.text[0] == '\0') || (event.text.text[1] != '\0'))`
and depend upon event.text.text actually being a NUL-terminated string.

**Comment by Vultraz** [comment]
`text` is apparently ```char[32] | text | the null-terminated input text in UTF-8 encoding```
.

**Comment by GregoryLundberg** [comment]
If it's UTF-8 encoded the test is wrong since it admits the invalid code-points \x80 through \xFF as a single-byte NUL-terminated string.

Use
`    ((event.text.text[0] == '\0') || (event.text.text[0] > '\x7F') || (event.text.text[1] != '\0'))`

**Comment by GregoryLundberg** [comment]
Probably should use `<= '\0'` instead of `== '\0'` so we're insensitive to signed char as well

**Comment by Vultraz** [comment]
Alright, I'll use that.

**Pull request merged**